addons/l10n_ph/models/res_partner.py

```diff
@@ -19,4 +20,20 @@ def _commercial_fields(self):
         return super()._commercial_fields() + ['branch_code']
 
     def check_vat_ph(self, vat):
-        return len(vat) == 17 and CHECK_VAT_PH_RE.match(vat)
+        return len(vat) == 11 and CHECK_VAT_PH_RE.match(vat)
+
+    @api.constrains('branch_code')
+    def check_branch_code(self):
+        if self.country_id.code != 'PH' or len(self.branch_code) == 3:
+            return
+        raise ValidationError(_(
+            'Branch Code %(branch_code)s does not respect the expected format %(format)s',
+            branch_code=self.branch_code,
+            format='123'
+        ))
+
+    def _get_name(self):
+        res = super()._get_name()
+        if self._context.get('show_vat') and self.vat and self.country_id.code == 'PH' and self.branch_code != '000':
+            res = '%s-%s' % (res, self.branch_code)
```

Comment:
What is someone inherits this for even other purposes?  Why not suppose that the branch code is part of the vat number?